packages/content/src/get.ts

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2020 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { IHubRequestOptions, getModel, getFromHubAPI } from "@esri/hub-common";
+// TODO: why won't this import from hub-common?
+// import { getFromHubAPI } from "../../common/src/api";
+
+import { IModel } from "@esri/hub-common";
+import { getCategory } from "@esri/hub-common";
+import { IContentModel } from "./index";
+
+/**
+ * ```js
+ * getContent('3ef...')
+ *  .then(contentModel => {
+ *    // work with the model
+ *  })
+ * ```
+ * Get the content by unique ID. 
+ * This method works to call either the Portal (AGO) API or Hub API
+ * and return a common data model.
+ *
+ *
+ * @param id - Content Id
+ * @param requestOptions - Tequest options that may have authentication manager
+ * @returns A Promise that will resolve with the Content metadata
+ * @export
+ */
+export function getContent(
+  id: string,
+  hubRequestOptions?: IHubRequestOptions
+): Promise<any> {
+
+  if(hubRequestOptions.isPortal) {
+    return _getContentFromPortal(id, hubRequestOptions);
+  } else {
+    return _getContentFromHub(id, hubRequestOptions);
+  }
+}
+
+/**
+ * ```js
+ * _getContentFromPortal('3ef...')
+ *  .then(contentModel => {
+ *    // work with the model
+ *  })
+ * ```
+ * Get the content data from Portal or Online 
+ *
+ *
+ * @param {String} id Content Id
+ * @param {IHubRequestOptions} hubRequestOptions Request options that may have authentication manager
+ * @returns A Promise that will resolve with the item and data
+ * @export
+ */
+export function _getContentFromPortal(
+  id: string,
+  hubRequestOptions?: IHubRequestOptions
+): Promise<any> {
+
+  return getModel(id, hubRequestOptions).then((item:IModel) => {
+    return new Promise(resolve => {
```

Comment:
so - either you can `const model = await getModel(...)` and then later just `return model` or you run iwth promises and start the chain w/ `return getModel(...).then(..next step...)....then(foo => { return foo;})`

Reply:
I'd prefer the `await` pattern as well.